src/odemis/driver/autoscript_client.py

```diff
@@ -0,0 +1,2065 @@
+        mag_range_min = self._hfw_nomag / fov_info["range"][1]
+        self.magnification = model.FloatContinuous(mag, unit="",
+                                                   range=(mag_range_min, mag_range_max),
+                                                   readonly=True)
+        # To provide some rough idea of the step size when changing focus
+        # Depends on the pixelSize, so will be updated whenever the HFW changes
+        self.depthOfField = model.FloatContinuous(1e-6, range=(0, 1e3),
+                                                  unit="m", readonly=True)
+        self._updateDepthOfField()
+
+        if has_detector:
+            rng = DETECTOR_RNG
+            self._shape = (rng[1][0], rng[1][1])
+            # pixelSize is the same as MD_PIXEL_SIZE, with scale == 1
+            # == smallest size/ between two different ebeam positions
+            pxs = (fov / self._shape[0],
+                   fov / self._shape[0])
+            # pixelsize is inferred indirectly via resolution and fov
+            self.pixelSize = model.VigilantAttribute(pxs, unit="m", readonly=True)
+
+            # scanning resolution
+            resolution = self.parent.get_resolution(self.channel)
+            res_choices = set(RESOLUTIONS)
+            self.resolution = model.VAEnumerated(resolution, res_choices, unit="px" , setter=self._setResolution)
+
+            # (float, float) as a ratio => how big is a pixel, compared to pixelSize
+            # it basically works the same as binning, but can be float.
+            # Defined as the scale to match the allowed resolutions, with pixels
+            # always square (ie, scale is always the same in X and Y).
+            scale = (self._shape[0] / resolution[0],) * 2
+            scale_choices = set((self._shape[0] / r[0],) * 2 for r in res_choices)
+            self.scale = model.VAEnumerated(scale, scale_choices, unit="", setter=self._setScale)
+            self.scale.subscribe(self._onScale, init=True)  # to update metadata
+
+            # Just to make some code happy
+            self.translation = model.TupleContinuous((0, 0), range=[(-512, -512), (512, 512)], unit="px", readonly=True)
+
+        # not yet implemented:
+        # beam on
+        # beam blank
+        # scanning mode
+        # spot size
+
+        # Refresh regularly the values, from the hardware, starting from now
+        self._updateSettings()
+        self._va_poll = util.RepeatingTimer(5, self._updateSettings, "Settings polling")
+        self._va_poll.start()
+
+    def _updateSettings(self) -> None:
+        """
+        Read all the current settings from the SEM and reflects them on the VAs
+        """
+        logging.debug("Updating SEM settings")
+        try:
+            dwell_time = self.parent.get_dwell_time(self.channel)
+            if dwell_time != self.dwellTime.value:
+                self.dwellTime._value = dwell_time
+                self.dwellTime.notify(dwell_time)
```

Comment:
Once you move the dwell time to "has_detector", this block must also be moved down into the "if has_detector" block.